File: src/main/java/kdh/boardproject/entity/BoardComment.java
package kdh.boardproject.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import kdh.boardproject.dto.boardComment.CreateBoardCommentDto;
import lombok.*;

import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.Date;

import static javax.persistence.FetchType.LAZY;

@Getter
@Setter
@Entity
@Table(name="board_comment")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class BoardComment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @JsonIgnore
    @Column(name="idx")
    private Long idx;

    @Column(name="content")
    private String content;

    @Column(name="createdAt")
    private LocalDateTime createdAt;

    @Column(name="updatedAt")
    private LocalDateTime updatedAt;

    @JsonIgnore
    @ManyToOne(fetch = LAZY)
    @JoinColumn(name = "created_user")
    private User user;

    @JsonIgnore
    @ManyToOne(fetch = LAZY)
    @JoinColumn(name = "board_idx")
    private Board board;

    @Builder
    public BoardComment(String content, User user, Board board){
        this.content = content;
        this.user = user;
        this.board = board;
        this.createdAt = LocalDateTime.now();
    }

}

File: src/main/java/kdh/boardproject/entity/BoardComment.kt
package kdh.boardproject.entity

import com.fasterxml.jackson.annotation.JsonIgnore
import lombok.*
import java.time.LocalDateTime
import javax.persistence.*


@Entity
@Table(name = "board_comment")
class BoardComment (
        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        @JsonIgnore
        @Column(name = "idx")
        val idx: Long? = null,

        @Column(name = "content")
        var content: String,

        @JoinColumn(name = "created_user")
        @ManyToOne(fetch = FetchType.LAZY)
        @JsonIgnore
        var user: User,

        @JoinColumn(name = "board_idx")
        @ManyToOne(fetch = FetchType.LAZY)
        @JsonIgnore
        var board: Board,

        @Column(name = "createdAt")
        var createdAt: LocalDateTime? = null,

        @Column(name = "updatedAt")
        var updatedAt: LocalDateTime? = null

        ) {

    init {
        this.createdAt = LocalDateTime.now()
    }
}